llama_stack/apis/telemetry/telemetry.py

```diff
@@ -123,10 +137,67 @@ class StructuredLogEvent(EventCommon):
 ]
 
 
+@json_schema_type
+class EvalTrace(BaseModel):
+    session_id: str
+    step: str
+    input: str
+    output: str
+    expected_output: str
+
+
+@json_schema_type
+class MaterializedSpan(Span):
+    children: List["MaterializedSpan"] = Field(default_factory=list)
+    status: Optional[SpanStatus] = None
+
+
+@json_schema_type
+class QueryCondition(BaseModel):
+    key: str
+    op: str
+    value: Any
+
+
+class TraceStore(Protocol):
```

Comment:
In this PR, we should at least move this out of apis/ and into either a util or an impl detail. Afterwards, we should be able to kill this perhaps. Also, we should not use OTel for writing to Sqlite or Console and do it directly (in the followup PR.)